### PR TITLE
Small tweaks/updates

### PR DESCRIPTION
**Pull request by donaldgray**
Couple of small tweaks around versioning:
* Add versions to requirements.txt
* Switch to `python:alpine` image for quicker build (final image size is very slightly smaller than before).

Fixed up the response, previously an exception could be thrown but it would always return `200 {"size": 1234, "success": True }.` This made it difficult for the calling code to know if it had failed (e.g. failed to upload the S3 object). Now `"success"` wil be `True|False` and will return non-2xx if fails.

Wrapped `POST /pdf` handler in `try..finally` and in finally remove the working directory. This lessens the need for external processes to tidy up working files.

Added a `GET /ping` endpoint for health-checks. Doesn't do much but could be extended if required.

Removed `POST /general-case/` endpoint as it contained an incomplete implementation.

Added defaults for all envvars to make it easier to quickly get up and running locally.

Added a github action to build + push to ghcr.

### CODE REVIEW

**Comment by donaldgray** [comment]
@tomcrane / @garyttierney are you happy for me to merge this? It's been running in spinup/wellcome and so far looks to be cleaning up files okay (although it's not been hit with a lot of traffic).

File: .github/workflows/.build.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    tags:
+      - "v*"
+  pull_request:
+
+jobs:
+  build-push:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Extract branch name for push
```
Comment (garyttierney) [inline .github/workflows/.build.yml]: You might prefer getting tags like this: https://github.com/digirati-co-uk/taxonomy-manager/blob/bugfix/ci/.github/workflows/ci.yml#L23-L38

**Review by garyttierney — APPROVED** [review]
> @tomcrane / @garyttierney are you happy for me to merge this? It's been running in spinup/wellcome and so far looks to be cleaning up files okay (although it's not been hit with a lot of traffic).

Left a minor comment on the CI workflow, but LGTM.

File: .github/workflows/.build.yml (hunk 1 above)
Comment (donaldgray) [reply .github/workflows/.build.yml]: Hmm, yeah so do I. Not sure where I must have copied this from an older action, will update.

**Pull request merged**